### PR TITLE
priorities apply to HTTP and may be used by extensions

### LINKED ISSUE #1550: How are DATAGRAM frames prioritized?
Discussed at the interim 2021-06.  We need to decide whether DATAGRAM is included and - if so - how.  If not, then the document might describe that it doesn't cover DATAGRAM.

### PR DESCRIPTION
**Pull request by LPardue**
Fixes #1550.

### CODE REVIEW

File: draft-ietf-httpbis-priority.md [hunk 1]
```diff
@@ -205,6 +205,17 @@ Where both endpoints disable HTTP/2 priorities, the client is expected to send
 this scheme's priority signal. Handling of omitted signals is described in
 {{parameters}}.
 
+# Applicability of the Extensible Priority Scheme
+
+The priority scheme defined by this document considers only the prioritization
+of HTTP response data. Clients express priority values to a server, which are
+input into scheduling decisions, see {{client-scheduling}} and
+{{server-scheduling}}.
```
Comment (kazuho) [inline draft-ietf-httpbis-priority.md:213]: Not that I have a good suggestion at this moment, but it seems to me that this text has two issues:
* It contradicts with {{client-scheduling}}, which says that a client might use the priority signal for scheduling "requests."
* Payload of successful CONNECT request is not HTTP response, but I think we are going to cover them in this specification? Assuming that is the case, prioritization is not limited to HTTP messages.

File: draft-ietf-httpbis-priority.md [hunk 2]
```diff
@@ -205,6 +205,17 @@ Where both endpoints disable HTTP/2 priorities, the client is expected to send
...
+
+Where HTTP extensions change stream behavior or define new data carriage
+mechanisms, they MAY also define how this priority scheme can be applied.
```
Comment (kazuho) [inline draft-ietf-httpbis-priority.md:216]: This paragraph LGTM.

**Review by kazuho — COMMENTED** [review]
Thank you for working on this.

File: draft-ietf-httpbis-priority.md (hunk 1 above)
Comment (LPardue) [reply draft-ietf-httpbis-priority.md:213]: Thanks, you're right on both counts. 

Over on https://github.com/httpwg/http-extensions/pull/1544/files, we're trying to accommodate the CONNECT method. So we could borrow some ideas. I'll follow up with a suggestion...

Comment (LPardue) [inline draft-ietf-httpbis-priority.md:213]:
```suggestion
The primary use case for the priority scheme defined by this document is the
prioritization of HTTP response data, which is transmitted as HTTP/2 or HTTP/3
DATA frames. Clients express priority values to a server using signals in the
form of a header field {{header-field}} or frame {{frame}}).  Signals are input
into scheduling decisions based on several factors (see {{merging}} and
{{server-scheduling}}). Clients can also use priority values for local
scheduling, see {{client-scheduling}}.
```

Comment (LPardue) [reply draft-ietf-httpbis-priority.md:213]: @kazuho does that address your comments?

Comment (kazuho) [reply draft-ietf-httpbis-priority.md:213]: Thank you for the suggestion. New text is technically correct though I wonder if we can condense the text, or remove references to HTTP/2, HTTP/3, considering the fact that we have been trying to be version-agnostic.

Maybe something as simple as: _The priority scheme defined by this document considers only the prioritization of HTTP messages and the payload of CONNECT method, see {{client-scheduling}}, {{server-scheduling}}, {{CONNECT}}._

Comment (LPardue) [inline draft-ietf-httpbis-priority.md:213]:
```suggestion
The priority scheme defined by this document considers only the prioritization
of HTTP messages and tunnels, see {{client-scheduling}}, {{server-scheduling}},
and {TODO: xref to CONNECT, depends on
https://github.com/httpwg/http-extensions/pull/1544}
```

Comment (LPardue) [reply draft-ietf-httpbis-priority.md]: Incorporated your simplified suggestion now, so resolving this thread

Comment (LPardue) [reply draft-ietf-httpbis-priority.md]: thanks, I've taken that!

**Review by kazuho — APPROVED** [review]

**Comment by LPardue** [comment]
I think solicit some feedback on this one to the WG before merging

**Review by DavidSchinazi — APPROVED** [review]

**Comment by LPardue** [comment]
Healthy discussion on list. On balance I'm merging this in the pursuit of progress.

File: draft-ietf-httpbis-priority.md (hunk 2 above)
Comment (martinthomson) [reply draft-ietf-httpbis-priority.md:216]: FWIW, I would prefer s/MAY/can/ - there is no normative effect of a choice.

Comment (LPardue) [reply draft-ietf-httpbis-priority.md:216]: missed this while tagging the release sorry. I'll create a followup ticket to roll into next round of editorial changes.

Comment (LPardue) [reply draft-ietf-httpbis-priority.md:216]: see https://github.com/httpwg/http-extensions/issues/1571

**Pull request merged**